### DISCUSSION TITLE
Return none if instance children is not loaded when using AsyncSession

### DISCUSSION THREAD

**Discussion by mockingjet** [sqlalchemy/sqlalchemy#6973] [Usage Questions] [answered]
Most Fastapi users like me would like to user AsyncSession to develop. And we usually take advantage of pydantic for data mapping. But sometimes it comes with some troubles when doing data mapping with sqlalchemy orm.

```py
# schema.py

from pydantic import BaseModel

class Book(BaseModel):
    id: int
    title: str

class User(BaseModel):
    id: str
    name: str
    books: Optional[list[BookModel]] # will try to access User.books when data mapping (even optional)
```
```py
# model.py

class BookModel(Base):
    id = sa.Column(sa.Integer, primary_key=True)
    title = sa.Column(sa.String, nullable=False)
    user_id = sa.Column(sa.ForeignKey("users.id"))

class UserModel(Base):
    id = sa.Column(sa.String, primary_key=True)
    name = sa.Column(sa.String, nullable=False)
    books = sa.orm.relationship("BookModel")
```

When I fetch a user from database without selectinload BookModel

```py
# api.py

@user_router.get(
    "/me",
    response_model=APIResponse[User], # data mapping here
)
async def get_me(
    current_user_id: str = Depends(get_current_user_id),
    user_service: UserSerivce = Depends(get_user_service),
):
    user = await user_service.get_user(current_user_id)  # not select with books

    return {"code": 0, "result": user}
```

It will throw error: **greenlet_spawn has not been called; can't call await_() here.**

Maybe change User schema can solve this problem. But what if a user has not just books but also addresses, login_records ... etc?

```py
# schema.py

from pydantic import BaseModel

class Book(BaseModel):
    id: int
    title: str

class User(BaseModel):
    id: str
    name: str

class UserWithBooks(User):
    books: list[BookModel]

class UserWithAddresses(User):
    addresses: list[AddressModel]

class UserWithLoginRecords(User):
    login_records: list[LoginRecordModel]

class UserWithBooksAndAddresses(UserWithBooks, UserWithAddresses):
    pass

# too many to list
```

So I wonder is there a quick solution or best practice to deal with this problem?

Maybe I can configure sqlalchemy async session or orm options, like when accessing **user.books**, it will return **None** if query user without selectinload BookModel

**Comment by CaselIT** [comment, marked as answer]
Hi,

Assuming the issue is in the loading of a relationship, you may be looking for 'noload'. See https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html for details

**Reply by CaselIT** [reply]
Regarding this

> So I wonder is there a quick solution or best practice to deal with this problem?

I'm not familiar enough with pydantinc to suggest anything. Ideally the serialization should serialized just what's loaded in the class, looking for them in `__dict__`, but I'm not sure if you can configure that with pydantic

**Reply by mockingjet** [reply]
To use **noload** as default

```py
books = sa.orm.relationship("BookModel", lazy=None)
```

This seems to solve my problem, thanks
